lib/configure.js

```diff
@@ -92,8 +92,22 @@ async function configure (gyp, argv) {
     log.verbose(
       'build dir', '"build" dir needed to be created?', isNew ? 'Yes' : 'No'
     )
-    const vsInfo = win ? await findVisualStudio(release.semver, gyp.opts['msvs-version']) : null
-    return createConfigFile(vsInfo)
+    if (win) {
+      let gypFormatIndex = argv.indexOf('-f')
+      if (gypFormatIndex === -1) {
+        gypFormatIndex = argv.indexOf('--format')
+      }
+      if (gypFormatIndex === -1 || !argv[gypFormatIndex + 1] || argv[gypFormatIndex + 1].startsWith('msvs')) {
+        const vsInfo = await findVisualStudio(release.semver, gyp.opts['msvs-version'])
+        return createConfigFile(vsInfo)
+      }
+      let vsInfo = {}
+      try {
+        vsInfo = await findVisualStudio(release.semver, gyp.opts['msvs-version'])
```

Comment:
I'm hesistant on ignoring the error here. I believe generators like ninja would depend on the settings as well.

Reply:
wasm case (specify `--format make-linux`) doesn't require Visual Studio. How can we distinguish between wasm and native here?

Reply:
Conservatively, I think excluding `make` generator would be sufficient. The make generator doesn't depend on the vs settings.